src/rust/inetstack/protocols/ip/protocol.rs

```diff
@@ -26,18 +24,20 @@ pub enum IpProtocol {
     UDP = 0x11,
 }
 
-//==============================================================================
+//======================================================================================================================
 // Trait Implementations
-//==============================================================================
+//======================================================================================================================
 
 /// TryFrom trait implementation.
 impl TryFrom<u8> for IpProtocol {
     type Error = Fail;
 
-    fn try_from(n: u8) -> Result<Self, Fail> {
-        match FromPrimitive::from_u8(n) {
-            Some(n) => Ok(n),
-            None => Err(Fail::new(ENOTSUP, "unsupported IPv4 protocol")),
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        match value {
+            0x01 => Ok(IpProtocol::ICMPv4),
```

Comment:
Same here, it would be nice to not just have these bare numbers with no explanation.

Reply:
Tese are defined in the enum itself, but I cannot access them in here. The function is precisely doing the conversion. Let me know if you know a different way of doing that in pure rust. The crates used before provide this "facility" at cost of non-standard 3rd party.

Reply:
Fixed